proposals/NativeInt.md

```diff
@@ -0,0 +1,264 @@
+
+In cases where there are two overloads, one for `nint` and one for `nuint`, the native int operand types are marked `native`. In other cases the specific types are provided.
+(The IL for each operator includes the variants for `nint` and `nuint` and the variants for `unchecked` and `checked` contexts if different.)
+
+| Unary | Operand | Result | IL |
+|:---:|:---:|:---:|:---:|
+| `-` | `nint` | `nint` | `neg` |
+| `~` | `native` | `native` | `not` |
+
+| Binary | Left | Right | Result | IL |
+|:---:|:---:|:---:|:---:|:---:|
+| `+` | `native` | `native` | `native` | `add` / `add.ovf` / `add.ovf.un` |
+| `-` | `native` | `native` | `native` | `sub` / `sub.ovf` / `sub.ovf.un` |
+| `*` | `native` | `native` | `native` | `mul` / `mul.ovf` / `mul.ovf.un` |
+| `/` | `native` | `native` | `native` | `div` / `div.un` |
+| `%` | `native` | `native` | `native` | `rem` / `rem.un` |
+| `==` | `native` | `native` | `native` | `beq` / `ceq` |
+| `!=` | `native` | `native` | `native` | `bne` |
+| `<` | `native` | `native` | `native` | `blt` / `clt` / `blt.un` / `clt.un` |
+| `<=` | `native` | `native` | `native` | `ble` / `ble.un` |
+| `>` | `native` | `native` | `native` | `bgt` / `cgt` / `bgt.un` / `cgt.un` |
+| `>=` | `native` | `native` | `native` | `bge` / `bge.un` |
+| `&` | `native` | `native` | `native` | `and` |
+| <code>&#124;</code> | `native` | `native` | `native` | `or` |
+| `^` | `native` | `native` | `native` | `xor` |
+| `<<` | `native` | `int` | `native` | `shl` |
+| `>>` | `native` | `int` | `native` | `shr` / `shr.un` |
+
+For some binary operators, the IL operators support additional operand types (see [ECMA-335](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-335.pdf)).
+But the set of operand types supported by C# is limited for simplicity and for consistency with existing operators in the language.
+
+Lifted versions of the operators, where the arguments and return types are `nint?` and `nuint?`, are supported.
+
+Compound assignment operations `x op= y` where `x` or `y` are native ints follow the same rules as with other primitive types with pre-defined operators.
+Specifically the expression is bound as `x = (T)(x op y)` where `T` is the type of `x` and where `x` is only evaluated once.
+
+### dynamic
+
+The conversions and operators are synthesized by the compiler and are not part of the underlying `IntPtr` and `UIntPtr` types.
+As a result those conversions and operators _are not available_ from the runtime binder for `dynamic`. 
+
+```C#
+nint x = 2;
+nint y = x + x; // ok
+dynamic d = x;
+nint z = d + x; // RuntimeBinderException: '+' cannot be applied 'System.IntPtr' and 'System.IntPtr'
+```
+
+### Type members
+
+The only constructor for `nint` or `nuint` is the parameter-less constructor.
+
+The members of `System.IntPtr` and `System.UIntPtr` other than the constructors and operators are available from `nint` and `nuint`.
```

Comment:
So all future members will automatically become available? That's a good thing. Maybe the spec should not say what is available but rather what members are replaced/added by synthetic members. All other members are just pass-through to `IntPtr`.